proto/axelarcork/v1/tx.proto

```diff
@@ -0,0 +1,50 @@
+syntax = "proto3";
+package axelarcork.v1;
+
+import "cosmos/base/v1beta1/coin.proto";
+import "axelarcork/v1/axelarcork.proto";
+import "gogoproto/gogo.proto";
+
+option go_package = "github.com/peggyjv/sommelier/v6/x/axelarcork/types";
+
+// MsgService defines the msgs that the cork module handles
+service Msg {
+  rpc ScheduleCork(MsgScheduleAxelarCorkRequest) returns (MsgScheduleAxelarCorkResponse);
+  rpc RelayCork(MsgRelayAxelarCorkRequest) returns (MsgRelayAxelarCorkResponse);
+  rpc BumpCorkGas(MsgBumpAxelarCorkGasRequest) returns (MsgBumpAxelarCorkGasResponse);
+}
+
+// MsgScheduleCorkRequest - sdk.Msg for scheduling a cork request for on or after a specific block height
+message MsgScheduleAxelarCorkRequest {
+  // the scheduled cork
+  AxelarCork cork = 1;
+  // the chain id
+  uint64 chain_id = 2;
+  // the block height that must be reached
+  uint64 block_height = 3;
+  // signer account address
+  string signer = 4;
+}
+
+message MsgScheduleAxelarCorkResponse {
+  // cork ID
+  string id = 1;
+}
+
+message MsgRelayAxelarCorkRequest {
+  string                   signer                  = 1;
+  cosmos.base.v1beta1.Coin token                   = 2 [(gogoproto.nullable) = false];
+  uint64                   fee                     = 3;
+  uint64                   chain_id                = 4;
+  string                   target_contract_address = 5;
```

Comment:
Noting the point here again: currently yes, per block height there can only be one cork per contract. However, let's say that there are calls X and Y at block heights 5 and 10. Both heights pass and both corks are approved, but have not been relayed yet. Since there is no cork ID or block height in a relay request, how could it tell the difference? Do we inspect in the relayer if there is more than one possible cork and make sure they happen in order?